### ISSUE TITLE
[Windows] PolyMC is not portable, always look for %APPDATA%

### ISSUE THREAD

**Issue by TetraTheta** [PolyMC/PolyMC#261]
### Operating System

Windows

### Description of bug

PolyMC's Windows latest build is `1.1.0-9f8983b5`. 

When I open PolyMC, it ignores `icons`, `instances`, `themes`, `translations`, `metacache`, `polymc.cfg` inside of its location - where `polymc.exe` locates - and re-create them in `%APPDATA%\PolyMC` directory.

### Steps to reproduce

  1. Prepare these files/directories already created before `1.1.0-9f8983b5`. Maybe you can use them in `%APPDATA%` instead, but I'm not sure about them. Backup these for if something goes wrong.
      * `icon`
      * `instances`
      * `themes`
      * `translations`
      * `metacache`
      * `polymc.cfg`
  2. Move those to where `polymc.exe` exists. For example, `polymc.cfg` and `polymc.exe` should be in same directory.
  3. Delete `%APPDATA%\PolyMC` if still exists.
  4. Launch PolyMC. It will now start wizard as if it is first launched.

### Suspected cause

In before, PolyMC(and MultiMC too) searched those files/directories at its 'installed' directory first. But after update, PolyMC now tries to search `%APPDATA%` for them. This is not what 'portable' means.

I think PolyMC should try to search those files/directories only when there is no `portable` blank file (for indicating it is portable mode) at `%APPDATA%`. 

PS. Setting directories manually in setting won't make it portable because `polymc.cfg` is still under `%APPDATA%`.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Scrumplex** [comment]
Not sure if we decided to not offer portable builds at all or if this is just a mistake on our side.

**Comment by LennyMcLennington** [comment]
> Not sure if we decided to not offer portable builds at all or if this is just a mistake on our side.

This is not intentional.

**Comment by DioEgizio** [comment]
yes but I prefer non-portable builds so please keep both

**Comment by ZekeZDev** [comment]
Having both is probably the best solution once we can figure out an installer though having an option in that to install system or user would be the best approach imo

**Comment by juliocoliveira** [comment]
That is a deal breaker for me, I need a portable version, not a fake portable. I don't want to loose all my mods, saves/worlds every time i format my windows. I'll be waiting this to be fixed.

**Comment by DioEgizio** [comment]
> That is a deal breaker for me, I need a portable version, not a fake portable. I don't want to loose all my mods, saves/worlds every time i format my windows. I'll be waiting this to be fixed.

If you really need it, for now you can build PolyMC with -DLauncher_PORTABLE=ON

**Comment by juliocoliveira** [comment]
Compile the whole launcher is not a option for me, Too complicated, I prefer wait for an official solution. For now i'll just use regular old MultiMC.

**Comment by TetraTheta** [comment]
> If you really need it, for now you can build PolyMC with -DLauncher_PORTABLE=ON

Since official build is broken, it is possible that code itself is somehow broken. If then, there is no way of user to compile PolyMC as portable.  
Correct me if I am wrong.

PS. I am also waiting fix for this, waiting while using MultiMC.

**Comment by DioEgizio** [comment]
> > If you really need it, for now you can build PolyMC with -DLauncher_PORTABLE=ON
> 
> Since official build is broken, it is possible that code itself is somehow broken. If then, there is no way of user to compile PolyMC as portable. Correct me if I am wrong.
> 
> PS. I am also waiting fix for this, waiting while using MultiMC.

No this is not true. The only thing is that there's no portable prebuilt build *yet*

**Comment by TetraTheta** [comment]
> No this is not true. The only thing is that there's no portable prebuilt build _yet_

I remember that I could use PolyMC as portable when early build of PolyMC, so I think something got broken between 1.1.0 and 1.0.0 (initial build of PolyMC).

**Comment by DioEgizio** [comment]
Nothing broke, there is just not *yet* a portable prebullt build

**Comment by TetraTheta** [comment]
How's that even possible? MultiMC which is 'origin' of the PolyMC already provided full portable build for a long time. If PolyMC is forked from MultiMC correctly, it should already has code or environment for building portable version of it because MultiMC already has it.

Also, MultiMC's Windows build was always portable.

TL;DR
1. As far as I remember, PolyMC was portable when its initial windows build.
2. Since origin of PolyMC, MultiMC already provided portable windows build for several years, PolyMC should also capable of providing portable windows build.

**Comment by DioEgizio** [comment]
> How's that even possible? MultiMC which is 'origin' of the PolyMC already provided full portable build for a long time. If PolyMC is forked from MultiMC correctly, it should already has code or environment for building portable version of it because MultiMC already has it.
> 
> Also, MultiMC's Windows build was always portable.
> 
> TL;DR
> 
>     1. As far as I remember, PolyMC was portable when its initial windows build.
> 
>     2. Since origin of PolyMC, MultiMC already provided portable windows build for several years, PolyMC should also capable of providing portable windows build.

Do you know what a *build* is?

**Comment by DioEgizio** [comment]
PolyMC *has* portable support but there's no build for it in the CI

**Comment by Scrumplex** [comment]
But we are actually trying to build a portable binary on Windows in the CI. There might be some issue in the code actually

**Comment by DioEgizio** [comment]
Oh sorry then

**Comment by Scrumplex** [comment]
Can someone provide their latest PolyMC log? Should be `PolyMC-0.log` in the data directory

**Comment by juliocoliveira** [comment]
I have this:
[PolyMC-0.log](https://github.com/PolyMC/PolyMC/files/8263783/PolyMC-0.log)

But is just a temp, I didn't extracted the files, just ran from winrar, and didn't launched the minecraft instance. Don't know if help much
